File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int left = 0;
        int right =0;
        Set<Character> track = new HashSet<>();
        int maxLen = 0;
        while(right< s.length()) {
            // Calculations
            // Increase Window Size
            if(!track.contains(s.charAt(right))) {
                track.add(s.charAt(right));
                maxLen = Math.max(maxLen, track.size());
                 right++;
            }

            else {
                // Slide from left 
                track.remove(s.charAt(left));
                left++;
            }
        }
        return maxLen;
        
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        

        character_set = set()
        left =0 
        max_length=0
        
        for right in range(len(s)):
            while(s[right] in character_set):
                character_set.remove(s[left])
                left+=1
            character_set.add(s[right])
            
            current_size = right-left+1

            max_length = max(current_size, max_length)


        return max_length



         










        